examples/chapter1/src/example3-cloner-jlbarros.cpp

```diff
@@ -0,0 +1,75 @@
+#include <algorithm>
+#include <iostream>
+#include <thread>
+
+#include <klepsydra/high_performance/event_loop_middleware_provider.h>
+
+const int poolSize{16};
+
+struct MyData
```

Comment:
This example is very simplistic. Use it to make it more relevant and didactic by converting this struct in a class with:
- constructor
- copy method
- init method
And use different combinations of publisher with and without `SmartObjectPool` to check the results.

Something like this:

```
class Event {
public:
static atomic<int> constructorCounter = 0;
static atomic<int> copyCounter = 0;
static atomic<int> initCounter = 0;

Event() {
   constructorCounter++;
}

Event(const Event & that) {
   this = that;
   copyCounter++;
}
void init () {
   initCounter++;
}

```

Reply:
ideas collected in [KCE-60](https://klepsydra.atlassian.net/browse/KCE-60), this branch will be removed.